content/en/docs/usage/gateway-api.md

```diff
@@ -0,0 +1,347 @@
+on Gateway resources. This works similarly as to what you can do with
+annotations on the Ingress resource, as described on the page [Securing Ingress
+Resources](/docs/usage/ingress/).
+
+The Gateway resource is part of the [Gateway API][gwapi], a set of CRDs that can
+be installed to your Kubernetes cluster and that aim at offering a richer
+alternative to the Ingress abstraction.
+
+[gwapi]: https://gateway-api.sigs.k8s.io
+
+The Gateway resource holds the TLS configuration, as illustrated in the
+following diagram (source: https://gateway-api.sigs.k8s.io):
+
+![Gateway vs. HTTPRoute](/images/gateway-roles.png)
+
+Note that cert-manager only supports setting up the TLS configuration on the
+Gateway resource when the Gateway is configured to terminate the TLS connection.
+There is still uncertainty around how the HTTPRoute TLS configuration should be
+used (see [the discussion about TLS interaction between HTTPRoute and
+Gateway][gateway-api#577]) and no existing implementation supports the TLS
+pass-through mode in which the HTTPRoute contains the TLS configuration and
+terminates the connection (see the [feature request for supporting `spec.tls` in
+HTTPRoute for Istio][istio#31747]).
+
+[istio#31747]: https://github.com/istio/istio/issues/31747
+[gateway-api#577]: https://github.com/kubernetes-sigs/gateway-api/issues/577
+
+Before enabling the Gateway support, you will have to install the Gateway API
+CRDs:
+
+```sh
+kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.3.0" | kubectl apply -f -
+```
+
+You then need to manually enable the Gateway support for Certificate using a
+flag:
+
+- If you are using Helm:
+
+  ```sh
+  helm upgrade --install cert-manager jetstack/cert-manager
+    --set "extraArgs={--controllers=*\,gateway-shim}"
+  ```
+
+- If you are using the raw cert-manager manifests, you can add the following
+  argument to the cert-manager controller Deployment's `args`:
+
+  ```yaml
+  args:
+    # ...
+    - --controllers=*,gateway-shim
+  ```
+
+You can also install the CRDs after installing cert-manager. cert-manager will
+pick up the Gateway CRD as soon as it is installed.
+
+The annotations `cert-manager.io/issuer` or `cert-manager.io/cluster-issuer`
+tells cert-manager that this Gateway should be looked at to create Certificate.
+For example, the following Gateway should trigger the creation of the
+`example-com-tls` Certificate:
+
+```yaml
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: Gateway
+metadata:
+  name: example
+  annotations:
+    cert-manager.io/issuer: foo
+spec:
+  gatewayClassName: foo
+  listeners:
+    - hostname: example.com
+      port: 443
+      protocol: HTTPS
+      routes:
+        kind: HTTPRoute
+        selector:
+          matchLabels:
+            app: foo
+      tls:
+        mode: Terminate
+        certificateRef:
+          name: example-com-tls
+          kind: Secret
+          group: core
```

Comment:
Will I be able to reference a Certificate at some point?

```suggestion
        certificateRef:
          name: example-com-tls
          kind: Certificate 
          group: cert-manager.io
```

Reply:
I guess this is up to the implementations of the Gatway API projects :)

Reply:
Yes. At the moment, `core/Secret` is the only resource that is supported across all Gateway API implementations. I read somewhere that [GKE Gateway Controller](https://cloud.google.com/kubernetes-engine/docs/concepts/gateway-api) plans on supporting another resource but I can't find where I read that.